### PR TITLE
Create a Lottery 

### PR DESCRIPTION
**Pull request by ArtemHurievskyi**
Create a Lottery that gives you 3 random balls with random colors and numbers.

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Color.values().length);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: let's move Random object creation to the class fields as it's currently created each time the method is called.

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+        Color color = Color.values()[index];
+        return color.toString();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]:
```suggestion
        return color.name();
```
Let's follow common mistakes

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Lottery extends Ball {
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:3]: invalid class inheritance. Let's not extend Ball here

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    private Ball ball;
+
+    public Ball getRandomBall() {
+        ball = new Ball();
+        getRandomNumber();
+        System.out.println(toString());
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:9]:
```suggestion
```

**Review by sarakhmen — CHANGES_REQUESTED** [review]
Let's improve your solution :)

File: src/main/java/core/basesyntax/Ball.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    private String color;
+    private int number;
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Ball.java:15]: Better use constructor instead of setters

File: src/main/java/core/basesyntax/Ball.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    public void getRandomNumber() {
+        int num = new Random().nextInt(100);
+        this.setNumber(num);
+    }
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Ball.java:20]: don`t use this method here

File: src/main/java/core/basesyntax/Lottery.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Lottery {
+    private Ball ball;
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Lottery.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    private ColorSupplier supplier;
+
+    public Ball getRandomBall() {
+        ball = new Ball();
+        ball.setColor(new ColorSupplier().getRandomColor());
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Lottery.java:9]:
```suggestion
        ball.setColor(colorSupplier.getRandomColor());
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Lottery {
+    private Ball ball;
+    private ColorSupplier supplier;
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Lottery.java:5]:
```suggestion
    private ColorSupplier supplier = new ColorSupplier();
```

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Application.java [hunk 10]
```diff
@@ -2,6 +2,10 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < 3; i++) {
+            Lottery lottery = new Lottery();
+            System.out.println(lottery.getRandomBall());
+        }
     }
+
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Application.java:10]:
```suggestion
```

File: src/main/java/core/basesyntax/Ball.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+    public String toString() {
+        return "The " + getRandomColor() + " ball with a " + number + " number!";
+    }
+
```
[the remark this answers is not in the document]
Comment (Vasul-Pankiv) [reply src/main/java/core/basesyntax/Ball.java:19]: not fixed

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 12]
```diff
@@ -1,7 +1,13 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private int index = new Random().nextInt(Color.values().length);
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: for what? This supplier always return the same value

File: src/main/java/core/basesyntax/Lottery.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private ColorSupplier supplier = new ColorSupplier();
+
+    public int getRandomNumber() {
+        int num = new Random().nextInt(100);
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Lottery.java:9]: make random a class level filed. Do not create Random obj  for each method call

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Application.java [hunk 14]
```diff
@@ -2,6 +2,9 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < 3; i++) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:5]: [All magic numbers in your code should be constants.](https://github.com/mate-academy/jv-lottery/blob/master/checklist.md#all-magic-numbers-in-your-code-should-be-constants)
make 3 as a constant

File: src/main/java/core/basesyntax/Lottery.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private final ColorSupplier supplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Ball getRandomBall() {
+        int num = random.nextInt(100);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:10]: make 100 as a constant

File: src/main/java/core/basesyntax/Lottery.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+        return new Ball(supplier.getRandomColor(), num);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:11]: create local variable for supplier.getRandomColor()

File: src/main/java/core/basesyntax/Lottery.java (hunk 15 above)
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:10]:
```suggestion
        int number = random.nextInt(100);
```

File: src/main/java/core/basesyntax/Application.java [hunk 17]
```diff
@@ -2,6 +2,9 @@
...
+            Lottery lottery = new Lottery();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:6]: [Don't create several instances of the same class if you can use only one instance for your purposes](https://github.com/mate-academy/jv-lottery/blob/master/checklist.md#dont-create-several-instances-of-the-same-class-if-you-can-use-only-one-instance-for-your-purposes)

Don't need create lottery object in loop

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Lottery.java [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int RANDOM_CEIL = 101;
+    private final ColorSupplier supplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Ball getRandomBall() {
+        return new Ball(supplier.getRandomColor(), random.nextInt(RANDOM_CEIL));
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:11]: It's better to create local variables for color and number and then provide to constructor

**Review by Elena-Bruyako — APPROVED** [review]
Approve, but pay attention to the comment

File: src/main/java/core/basesyntax/Lottery.java (hunk 18 above)
Comment (ArtemHurievskyi) [reply src/main/java/core/basesyntax/Lottery.java:11]: Thank you for your advice, I appreciate this!

**Pull request closed without merging**